internal/compiler-interface/src/main/java/xsbti/compile/IncrementalCompiler.java

```diff
@@ -37,32 +38,73 @@
+     *
+     * @return An instance of {@link CompileResult} that holds information
+     * about the results of the compilation.
+     */
+    CompileResult compile(Inputs inputs, Logger logger);
 
-  /**
-   * Creates a compiler instance that can be used by the `compile` method.
-   *
-   * @param instance The Scala version to use
-   * @param interfaceJar The compiler interface jar compiled for the Scala version being used
-   * @param options Configures how arguments to the underlying Scala compiler will be built.
-   */
-  // ScalaCompiler newScalaCompiler(ScalaInstance instance, File interfaceJar, ClasspathOptions options);
+    /**
+     * Performs an incremental compilation given its configuration.
+     *
+     * @param scalaCompiler The Scala compiler to compile Scala sources.
+     * @param javaCompiler The Java compiler to compile Java sources.
+     * @param sources An array of Java and Scala source files to be compiled.
+     * @param classpath An array of files representing classpath entries.
+     * @param output An instance of {@link Output} to store the compiler outputs.
+     * @param globalsCache Directory where previous cached compilers are stored.
+     * @param scalacOptions An array of options/settings for the Scala compiler.
+     * @param javacOptions An array of options for the Java compiler.
+     * @param previousAnalysis Optional previous incremental compilation analysis.
+     * @param previousSetup Optional previous incremental compilation setup.
+     * @param perClasspathEntryLookup Lookup of data structures and operations
+     *                                for a given classpath entry.
+     * @param reporter An instance of {@link Reporter} to report compiler output.
+     * @param compileOrder The order in which Java and Scala sources should
+     *                     be compiled.
+     * @param skip Flag to ignore this compilation run and return previous one.
+     * @param progress An instance of {@link CompileProgress} to keep track of
+     *                 the current compilation progress.
+     * @param incrementalOptions An Instance of {@link IncOptions} that
+     *                           configures the incremental compiler behaviour.
+     * @param extra An array of sbt tuples with extra options.
+     * @param logger An instance of {@link Logger} that logs Zinc output.
+     *
+     *
+     * @return An instance of {@link CompileResult} that holds information
+     * about the results of the compilation.
+     */
+    CompileResult compile(ScalaCompiler scalaCompiler,
+                          JavaCompiler javaCompiler,
+                          File[] sources,
+                          File[] classpath,
+                          Output output,
+                          GlobalsCache globalsCache,
+                          String[] scalacOptions,
+                          String[] javacOptions,
+                          Maybe<CompileAnalysis> previousAnalysis,
```

Comment:
@eed3si9n Are you sure you want to make that commitment? We also compile to 2.11, so the folks that want to use that won't be able to run it on JDK7. Honestly, I think that a lot of people are stuck in it yet.
  
What I propose is that we keep this out of this PR and discuss it separately in another issue.